packages/async-flow/test/async-flow-debug.test.js

```diff
@@ -0,0 +1,219 @@
+  // durable object state are noisy and spurious.
+  for (let i = 0; i < recLen17; i += 1) {
+    if (expectedRecording[i] === undefined) {
+      expectedRecording[i] = consSentAs(i);
+    }
+  }
+  t.like(recording, expectedRecording);
+  t.is(Object.keys(indexToStackRegex).length, 0, 'all stack entries tested');
+};
+
+/**
+ * @param {Zone} zone
+ * @param {number} [k]
+ */
+const prepareOrchestra = (zone, k = 1) =>
+  zone.exoClass(
+    'Orchestra',
+    undefined,
+    (factor, vow, resolver) => ({ factor, vow, resolver }),
+    {
+      scale(n) {
+        const { state } = this;
+        return k * state.factor * n;
+      },
+      vow() {
+        const { state } = this;
+        return state.vow;
+      },
+      resolve(x) {
+        const { state } = this;
+        state.resolver.resolve(x);
+      },
+      reject(x) {
+        const { state } = this;
+        state.resolver.reject(x);
+      },
+    },
+  );
+
+/** @typedef {ReturnType<ReturnType<prepareOrchestra>>} Orchestra */
+
+/**
+ * @param {ExecutionContext<unknown>} t
+ * @param {Zone} zone
+ */
+const testRejectionPlay = async (t, zone) => {
+  t.log('rejectionPlay started');
+  const vowTools = prepareVowTools(zone);
+  const { asyncFlow, adminAsyncFlow } = prepareTestAsyncFlowTools(t, zone, {
+    vowTools,
+  });
+  const makeOrchestra = prepareOrchestra(zone);
+  const { makeVowKit } = vowTools;
+
+  const { vow: v2, resolver: r2 } = zone.makeOnce('v2', () => makeVowKit());
+  const hOrch7 = zone.makeOnce('hOrch7', () => makeOrchestra(7, v2, r2));
+
+  // purposely violate rule that guestMethod is closed.
+  const { promise: promiseStep, resolve: resolveStep } = makePromiseKit();
+
+  const { guestMethod } = {
+    async guestMethod(gOrch7) {
+      const p2 = gOrch7.vow();
+
+      await eventLoopIteration();
+
+      const caughtP = p2.catch(e => {
+        try {
+          // Capture stack details to the console.
+          consoleRecorder.go();
+          console.error('p2 rejection:', e);
+        } finally {
+          rejectionRecordingAssertions(t, hOrch7, e, consoleRecorder.stop());
+        }
+      });
+
+      setImmediate(() => {
+        gOrch7.reject(Error('should be seen'));
+      });
+
+      await eventLoopIteration();
+      resolveStep(caughtP);
+      return 'done';
+    },
+  };
+
+  const wrapperFunc = asyncFlow(zone, 'AsyncFlow1', guestMethod);
+
+  const outcomeV = zone.makeOnce('outcomeV', () => wrapperFunc(hOrch7));
+
+  t.true(isVow(outcomeV));
+
+  const flow = zone.makeOnce('flow', () =>
+    adminAsyncFlow.getFlowForOutcomeVow(outcomeV),
+  );
+  t.is(passStyleOf(flow), 'remotable');
+
+  await promiseStep;
+  t.is(await vowTools.when(outcomeV), 'done');
+
+  const logDump = flow.dump();
+  t.is(logDump.length, 0);
+  t.log('rejectionPlay done');
+};
+
+test.serial('test host vow rejection logging', async t => {
+  annihilate();
+  const zone1 = makeDurableZone(getBaggage(), 'durableRoot');
+  await testRejectionPlay(t, zone1);
+});
```

Comment:
I think I'd like to see this test also written with a restart where the vow is only resolved in the new incarnation.